### ISSUE TITLE
Invalid argument: The graph couldn't be sorted in topological order

### ISSUE THREAD

**Issue by yezhengli-Mr9** [GiaLacTRAN/convolutional_deep_gp_random_features#1]
For python 3.6, tf1.12, I will confront following
`iterations:0, err_test:0.9026998901367187, mnll(loss)_test:2.4577505493164065, ece_test:0.1034441037118435, brier_test:0.9321980105054014, train_time:0, test_time:41433
2020-08-09 00:56:36.970979: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:533] remapper failed: Invalid argument: MutableGraphView::SortTopologically error: detected edge(s) creating cycle(s) {'mul_277' -> 'gradients/add_105_grad/tuple/control_dependency'}.
2020-08-09 00:56:36.972033: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:533] arithmetic_optimizer failed: Invalid argument: The graph couldn't be sorted in topological order.
2020-08-09 00:56:36.972945: E tensorflow/core/grappler/optimizers/dependency_optimizer.cc:697] Iteration = 0, topological sort failed with message: The graph couldn't be sorted in topological order.
2020-08-09 00:56:36.973840: E tensorflow/core/grappler/optimizers/dependency_optimizer.cc:697] Iteration = 1, topological sort failed with message: The graph couldn't be sorted in topological order.`

whatever via (I reduce batch size slightly)
`python main.py --data_name=cifar10 --cnn_name=lenet3conv --rf_name=rf --nb_gp_blocks=5 --ratio_nrf_df=1 --train_time=21600000 --display_time=3600000 --ratio_train_size=1.0 --test_size=1000 --train_batch_size=16 --test_batch_size=100 --learning_rate=0.001 --mc_test=50 --num_bins=20 --less_print=True`
or
`python main.py --data_name=mnist --cnn_name=mnistlenet --rf_name=rf --nb_gp_blocks=1 --ratio_nrf_df=1 --train_time=21600000 --display_time=3600000 --ratio_train_size=1.0 --test_size=10000 --train_batch_size=16 --test_batch_size=16 --learning_rate=0.001 --mc_test=50 --num_bins=20 --less_print=True`
whatever with or without GPU/ CUDA.
